easybuild/tools/systemtools.py

```diff
@@ -160,24 +161,35 @@ class SystemToolsException(Exception):
 def sched_getaffinity():
     """Determine list of available cores for current process."""
     cpu_mask_t = ctypes.c_ulong
-    cpu_setsize = 1024
     n_cpu_bits = 8 * ctypes.sizeof(cpu_mask_t)
-    n_mask_bits = cpu_setsize // n_cpu_bits
-
-    class cpu_set_t(ctypes.Structure):
-        """Class that implements the cpu_set_t struct."""
-        _fields_ = [('bits', cpu_mask_t * n_mask_bits)]
 
     _libc_lib = find_library('c')
-    _libc = ctypes.cdll.LoadLibrary(_libc_lib)
+    _libc = ctypes.CDLL(_libc_lib, use_errno=True)
 
     pid = os.getpid()
-    cs = cpu_set_t()
-    ec = _libc.sched_getaffinity(os.getpid(), ctypes.sizeof(cpu_set_t), ctypes.pointer(cs))
-    if ec == 0:
-        _log.debug("sched_getaffinity for pid %s successful", pid)
-    else:
-        raise EasyBuildError("sched_getaffinity failed for pid %s ec %s", pid, ec)
+
+    cpu_setsize = 1024  # Max number of CPUs currently detectable
+    max_cpu_setsize = ctypes.c_ulong(-1).value // 4  # (INT_MAX / 2)
```

Comment:
You could do max_cpu_setsize = cpu_mask_t(-1).value // 4 for consistency. And why the "//" instead of simply "/", asking since I don't know python enough.

Reply:
`//` is integer division in Python 3, see https://riptutorial.com/python/example/2797/integer-division.

It's the correct way to ensure you get an integer both in Python 2 and 3, no float surprises :)

In Python 3:

```
>>> 3/2
1.5
>>> 3//2
1
```

Reply:
good to know, learning by pestering those who know better :-)

Reply:
fixed in 3c1279e